File: 0241-17-3394-check-if-grid-can-be-cut-into-sections/0241-17-3394-check-if-grid-can-be-cut-into-sections.py
class Solution:
    def checkValidCuts(self, n: int, rectangles: List[List[int]]) -> bool:
        def check(rectangles,d):
            rectangles.sort(key=lambda rectangles:rectangles[d])

            cnt,mx = 0 ,rectangles[0][d+2]
            for i in range(1,len(rectangles)):
                if mx <= rectangles[i][d]:
                    cnt+=1
                mx = max(mx,rectangles[i][d+2])
            return cnt >= 2
        
        return check(rectangles,0) or check(rectangles,1)

File: 0241-17-3394-check-if-grid-can-be-cut-into-sections/0241-17-3394-check-if-grid-can-be-cut-into-sections.java

class Solution {
    private boolean check(int[][] rectangles, int d) {
        Arrays.sort(rectangles, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                return Integer.compare(a[d], b[d]);
            }
        });
        
        int cnt = 0;
        int mx = rectangles[0][d + 2]; 
        for (int i = 1; i < rectangles.length; i++) {
            if (mx <= rectangles[i][d]) {
                cnt++;
            }
            mx = Math.max(mx, rectangles[i][d + 2]);
        }
        return cnt >= 2;
    }

    public boolean checkValidCuts(int n, int[][] rectangles) {
        return check(rectangles, 0) || check(rectangles, 1);
    }
}